frontend/src/components/auth/LoginForm.tsx

```diff
@@ -0,0 +1,148 @@
+import React from "react";
+import {
+  Checkbox,
+  Flex,
+  FormControl,
+  FormHelperText,
+  FormLabel,
+  IconButton,
+  Input,
+  InputGroup,
+  InputRightElement,
+  Text,
+} from "@chakra-ui/react";
+import { FaRegEye, FaRegEyeSlash } from "react-icons/fa";
+
+type LoginFormProps = {
+  isSignup: boolean;
+  invalidLogin: boolean;
+  firstName: string;
+  setFirstName: (firstName: string) => void;
+  lastName: string;
+  setLastName: (lastName: string) => void;
+  email: string;
+  setEmail: (email: string) => void;
+  password: string;
+  setPassword: (password: string) => void;
+  agreeToTerms: boolean;
+  onAgreeToTermsClick: () => void;
+};
+
+const LoginForm = ({
+  isSignup,
+  invalidLogin,
+  firstName,
+  setFirstName,
+  lastName,
+  setLastName,
+  email,
+  setEmail,
+  password,
+  setPassword,
+  agreeToTerms,
+  onAgreeToTermsClick,
+}: LoginFormProps) => {
+  const [showPassword, setShowPassword] = React.useState(false);
+
+  const onShowPasswordClick = async () => {
+    setShowPassword(!showPassword);
+  };
+
+  // TODO: Display terms and conditions.
+  const onTermsClick = () => {
+    console.log("Display terms and conditions.");
+  };
+
+  return (
+    <Flex direction="column">
+      {isSignup && (
+        <Flex direction="column">
+          <FormControl isRequired marginTop="30px">
+            <FormLabel htmlFor="first-name">First name</FormLabel>
+            <Input
+              id="first-name"
+              size="md"
+              type="text"
+              value={firstName}
+              onChange={(event) => setFirstName(event.target.value)}
+              placeholder="Enter first name"
+            />
+          </FormControl>
+          <FormControl isRequired marginTop="30px">
+            <FormLabel htmlFor="last-name">Last name</FormLabel>
+            <Input
+              id="last-name"
+              size="md"
+              type="text"
+              value={lastName}
+              onChange={(event) => setLastName(event.target.value)}
+              placeholder="Enter last Name"
+            />
+          </FormControl>
+        </Flex>
+      )}
+      <FormControl isRequired={isSignup} marginTop="30px">
+        <FormLabel htmlFor="email">Email Address</FormLabel>
```

Comment:
nit: in the invalid stage, the mockups have red text for field labels
<img width="398" alt="Screen Shot 2021-10-21 at 10 07 17 PM" src="https://user-images.githubusercontent.com/37675216/138381683-9e86aeaa-e580-4f15-879b-3870ee149399.png">